cumulus_library/template_sql/shared_macros/syntax.sql.jinja

```diff
@@ -17,4 +17,16 @@
 UNION
 {%- endif -%}
 {%- endmacro -%}
+
+{#- The intent of this function is to allow an end user to work in SQL for partial
+string searches, rather than regex. So for example, a value of macro_string like:
+
+https://fhir.cerner.com/%/codeSet/71
+
+will be converted into a regex-escaped pattern with a regex match anything pattern:
+
+https:\/\/fhir\.cerner\.com\/(.*)\/codeSet\/71 -#}
+{%- macro wildcard_to_regex(match_string) -%}
+{{ match_string.replace("/", "\\/").replace(".", "\\.").replace("%", "(.*)") }}
+{%- endmacro -%}
```

Comment:
- You must add `^` and `$` at ends of the resulting string to match wildcard semantics (`LIKE` is a direct match, `regexp_like()` is a search).
- Why are we escaping forward slashes? Those don't seem to be part of the regex syntax.
- Though we should probably be escaping backslashes, dollar signs, and a whole bunch of stuff if you're paranoid. But since we control the source patterns (so far), we can get away without it. It's just annoying that trino doesn't provide a "make this string a regex literal kthx" method.
- Why the parens around `.*`? - I'm used to that being used for grouping but you're not referring back to the group here.
- Totes optional, but you could return an actual string (i.e. add `'` around the result) so every caller doesn't have to do that.

Reply:
1,4,5 - makes sense, will do.
2 - well i was looking at this in a regex debugger and apparently had PHP syntax on, which was the default, which... anyway, it seems to work both ways, but i'll change it.
3 - We could go crazy on this one, but yeah, since this is code systems right now, i scoped to fit, but if there's something you're feeling paranoid about, now's as good a time as any.

Reply:
Naw I don't feel strongly on 3 - though....  we should almost certainly escape the single quote character `'`. I think as long as you do that, we should avoid truly disastrous outcomes and merely have the wrong matches if a string goes sideways.

Reply:
Or mabye jinja auto-escapes quotes like that for us, I forget.